resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/MediaTypeMap.java

```diff
@@ -272,7 +272,7 @@ private CachedMediaTypeAndClass(Class clazz, MediaType mediaType)
       {
          this.clazz = new WeakReference(clazz);
          this.mediaType = mediaType;
-         int result = getClazz().hashCode();
+         int result = clazz.hashCode();

```

Comment:
Yes. The hash code of the reference object is not the same of the Class that's referenced within the reference. CachedMediaTypeAndClass seems to be designed to that its hash code is the same of the Class that's passed when building it. And the reason for that is most likely related to fact that CachedMediaTypeAndClass instances are being added to a hashmap. By using the hash code of every new WeakReference that's built when creating a new CachedMediaTypeAndClass, you would basically alter the behavior.